src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,417 @@
+package duke;
+
+import duke.taskexceptions.EmptyTaskDateException;
+import duke.taskexceptions.NoTaskDateException;
+import duke.taskexceptions.NoTaskNameException;
+import duke.taskexceptions.TaskDateFormatException;
+
+import duke.tasksmanager.*;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.FileWriter;
+import java.io.PrintWriter;
+
+import java.util.ArrayList;
+import java.util.Scanner;
+
+public class Duke {
+
+    public static String line = "____________________________________________________________";
+
+    public static ArrayList<Tasks> tasks = new ArrayList<>(); //for storing (all types of) tasks
+    public static int taskCount = 0; //for counting tasks
+    public static String taskInputString; //contains taskName and taskDate (from user's input)
+    public static String taskName;
+    public static String taskDate;
+
+    public static File dukePrevInputList = new File("data/dukeList.txt");
+    // creates a File object to represent a file duke.txt
+    // that exists in the data directory relative to the current working directory
+
+    /**
+     * Prints some lines to welcome the user:
+     */
+    public static void saysHiToUser() {
+        //Greeting:
+        System.out.println(line);
+        System.out.println("Hello! I'm Duke");
+        System.out.println("What can I do for you?");
+        System.out.println(line);
+    }
+
+    /**
+     * Prints some lines when user exits:
+     */
+    public static void saysByeToUser() {
+        //exits with "bye":
+        System.out.println(line);
+        System.out.println("Bye. Hope to see you again soon!"); //exits
+        System.out.println(line);
+    }
+
+    /**
+     * Takes in User's input string
+     * Adds todo type task to 'tasks' array
+     * Prints added todo task
+     *
+     * @param input - user's input string
+     */
+    public static void addToDo(String input) {
+        try {
+            separateTypeOfTaskAndTaskInputString(input);
+            taskName = taskInputString;
+            tasks.add(new ToDos(taskName)); //add task to list
+
+            System.out.println("Got it. I've added this task:");
+            printAddedTask();
+
+        } catch (NoTaskNameException e) {
+            System.out.println("☹ OOPS!!! The description of a todo cannot be empty.");
```

Comment:
You have some methods that abstracted away details. Consider if [SLAP](https://nus-cs2113-ay2021s2.github.io/website/se-book-adapted/chapters/codeQuality.html#slap-hard) could work here.